core/src/main/scala/org/apache/spark/api/python/PythonRDD.scala

```diff
@@ -889,21 +891,40 @@ private class PythonAccumulatorParam(@transient private val serverHost: String,
     socket
   }
 
-  override def zero(value: JList[Array[Byte]]): JList[Array[Byte]] = new JArrayList
+  override def reset(): Unit = {
+    this._acc = new JArrayList[Array[Byte]]
+  }
+
+  override def isZero: Boolean = {
+    this._acc.isEmpty
+  }
 
-  override def addInPlace(val1: JList[Array[Byte]], val2: JList[Array[Byte]])
-      : JList[Array[Byte]] = synchronized {
+  override def copy(): PythonAccumulatorV2 = {
+    val newAcc = new PythonAccumulatorV2(serverHost, serverPort)
+    newAcc._acc = this._acc
```

Comment:
Yah `copy()` could be used elsewhere besides inside of `copyAndReset()` - although I don't see any blindingly obvious places where it is used inside of Spark. I'll switch this to be more like the `CollectionAccumulator` and have a specialized `copyAndReset` and do a deep copy on `copy`.